src/Core/Framework/Adapter/Twig/SwTwigFunction.php

```diff
@@ -38,9 +38,11 @@ class SwTwigFunction
     public static function getAttribute(Environment $env, Source $source, mixed $object, mixed $item, array $arguments = [], $type = /* Template::ANY_CALL */ 'any', $isDefinedTest = false, $ignoreStrictCheck = false, $sandboxed = false, int $lineno = -1)
     {
         try {
+            FieldVisibility::$isInTwigRenderingContext = true;
+            if ($object instanceof Markup && self::$macroResult !== null && $type === 'array') {
```

Comment:
quick idea: maybe we don't need to have this callMacro hack, but here we can look if we have a macroResult (as is done in the return node) and if that is defined we can use that, no need to check type

only thing is we would need to reset the result here when we used it once


```php
if ($object instanceof Markup && self::$macroResult !== null) {
   $object = self::$macroResult;
   self::$macroResult = null;
}
```